### PR TITLE
feat: allow for more extensive benchmark file in jsonl format

### PR DESCRIPTION
**Pull request by fgvieira**
### Description

Another try at #2176, where we try to gather more information about a job; for example, not only resources used, but also resources requested and size of input files.
Here, the original `benchmark` file is left as is, but there is an option to request an extended `benchmark` file if the extension is `.jsonl`.

Is there any other relevant info to include?

@SichongP @corneliusroemer feel free to comment.

### QC

* [x] The PR contains a test case for the changes or the changes are already covered by an existing test case.
* [x] The documentation (`docs/`) is updated to reflect the changes or this is not necessary (e.g. if the change does neither modify the language nor the behavior or functionalities of Snakemake).

### CODE REVIEW

**Comment by SichongP** [comment]
Is this a replacement of my PR? Sorry I haven't looked much into it. Just want to know if I still need to refactor my PR to the new format (I should have some more time to work on it in the next two weeks)

**Comment by fgvieira** [comment]
Not sure... I started it just to explore a simpler way to do it (just adding more info to the `benchmark` file), but I don't know if this is the best way to go (hence all the questions).

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Great work!

File: docs/snakefiles/rules.rst [hunk 1]
```diff
@@ -1931,11 +1931,25 @@ With the `benchmark` keyword, a rule can be declared to store a benchmark of its
...
+* `io_out`: I/O written (in bytes),
+* `mean_load`: CPU load = CPU time (`cpu_usages`) divided by wall clock time (`s`),
+* `cpu_time`: CPU time user+system (seconds),
+
+Since version X.X, it is possible to have extra benchmark metrics with the command ``--benchmark-extended``:
+* `jobid`: Internal job ID,
+* `rule_name`: Name of rule,
+* `wildcards`: Wildcards of this job,
+* `threads`: Number of threads requested for this job,
+* `cpu_usages`: Total CPU time (seconds),
```
[the remark this answers is not in the document]
Comment (fgvieira) [reply docs/snakefiles/rules.rst:1950]: Those values are collected by `benchmarks` and I have not added any new, but `cpu_time` is the total user+system time, while `cpu_usage` seemd to be the CPU load on a period of time.

https://github.com/snakemake/snakemake/blob/f55801bfabfb916f77fa4ec8f9eb0667e5c8a92e/snakemake/benchmark.py#L231-L234

**Review by johanneskoester — APPROVED** [review]

File: docs/snakefiles/rules.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanneskoester) [reply docs/snakefiles/rules.rst]: Ok, but shouldn"t this then be called cpu_usage and not cpu_usages?

[the remark this answers is not in the document]
Comment (fgvieira) [reply docs/snakefiles/rules.rst]: sure, I can change it.

**Pull request merged**